sdk/src/Core/Amazon.Runtime/Credentials/FederatedAWSCredentials.cs

```diff
@@ -195,9 +255,79 @@ private CredentialsRefreshState Authenticate(ICredentials userCredential)
+            catch (Exception e)
+            {
+                var wrappedException =
+                    new AmazonClientException("Credential generation from SAML authentication failed.", e);
+
+                var logger = Logger.GetLogger(typeof(FederatedAWSCredentials));
+                logger.Error(wrappedException, wrappedException.Message);
+
+                throw wrappedException;
+            }
+
+            return state;
+        }
+
+        private async Task<CredentialsRefreshState> AuthenticateAsync(ICredentials userCredential)
+        {
+            CredentialsRefreshState state;
+
+            var region = Options.STSRegion;
+            if (region == null)
+            {
+                region = FallbackRegionFactory.GetRegionEndpoint();
+            }
+            if (region == null)
+            {
+                region = DefaultSTSClientRegion;
+            }
+
+            ICoreAmazonSTS coreSTSClient = GetSTSClient(region);
+
+            try
+            {
+                var credentials = await coreSTSClient.CredentialsFromSAMLAuthenticationAsync(
+                    SAMLEndpoint.EndpointUri.ToString(),
+                    SAMLEndpoint.AuthenticationType.ToString(), RoleArn, MaximumCredentialTimespan, userCredential).ConfigureAwait(false);
+
+                RegisterRoleSession(credentials);
+
+                state = new CredentialsRefreshState(credentials, credentials.Expires);
+            }
+            catch (Exception e)
+            {
+                var wrappedException =
+                    new AmazonClientException("Credential generation from SAML authentication failed.", e);
+
+                var logger = Logger.GetLogger(typeof(FederatedAWSCredentials));
+                logger.Error(wrappedException, wrappedException.Message);
+
+                throw wrappedException;
+            }
+
+            return state;
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026",
+           Justification = "Reflection code is only used as a fallback in case the SDK was not trimmed. Trimmed scenarios should register dependencies with Amazon.RuntimeDependencyRegistry.GlobalRuntimeDependencyRegistry")]
+        private ICoreAmazonSTS GetSTSClient(RegionEndpoint region)
+        {
             ICoreAmazonSTS coreSTSClient = GlobalRuntimeDependencyRegistry.Instance.GetInstance<ICoreAmazonSTS>(ServiceClientHelpers.STS_ASSEMBLY_NAME, ServiceClientHelpers.STS_SERVICE_CLASS_NAME,
-                new CreateInstanceContext(new SecurityTokenServiceClientContext { Action = SecurityTokenServiceClientContext.ActionContext.AssumeRoleAWSCredentials, Region = region, ProxySettings = Options?.ProxySettings }));
-            if(coreSTSClient == null)
+                            new CreateInstanceContext(new SecurityTokenServiceClientContext { Action = SecurityTokenServiceClientContext.ActionContext.AssumeRoleAWSCredentials, Region = region, ProxySettings = Options?.ProxySettings }));
```

Comment:
Nit: Remove extra space before `new CreateInstance...` (this happens in `sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/EndpointDiscoveryHandler.cs` too).

Reply:
Fixed in latest commit https://github.com/aws/aws-sdk-net/pull/3681/commits/e9ede6c25ed4a3e7e2ff9ea18b0ce7ccefcf7750. Will initiate a dry run.